res/controllers/common-hid-packet-parser.js

```diff
@@ -415,6 +424,14 @@ HIDPacket.prototype.addControl = function(group, name, offset, pack, bitmask, is
 
     // Add the new field to the packet
     control_group[field.id] = field
+    
+    if (callback !== undefined) {
+        if (typeof callback === "function") {
```

Comment:
this is a nit, but putting exception clauses first makes for neater code down the line.  I realize my example may not have followed that rule:
```
if (callback === undefined) {
   return;
}
if (typeof callback !== "function") {
  HIDDebug("ERROR..."
  return;
}
this.setCallback(group, name, callback);
```